Gems/Terrain/Code/Source/TerrainRenderer/Components/TerrainSurfaceMaterialsListComponent.h

```diff
@@ -25,18 +25,24 @@ namespace LmbrCentral
 
 namespace Terrain
 {
-    struct TerrainSurfaceMaterialMapping final
+    struct TerrainSurfaceMaterialMapping
     {
-    public:
         AZ_CLASS_ALLOCATOR(TerrainSurfaceMaterialMapping, AZ::SystemAllocator, 0);
         AZ_RTTI(TerrainSurfaceMaterialMapping, "{37D2A586-CDDD-4FB7-A7D6-0B4CC575AB8C}");
         static void Reflect(AZ::ReflectContext* context);
 
-        SurfaceData::SurfaceTag m_surfaceTag;
-        AZ::Data::AssetId m_activeMaterialAssetId;
+        virtual ~TerrainSurfaceMaterialMapping() = default;
+
         AZ::Data::Asset<AZ::RPI::MaterialAsset> m_materialAsset;
         AZ::Data::Instance<AZ::RPI::Material> m_materialInstance;
 
+        AZ::Data::AssetId m_activeMaterialAssetId;
+        AZ::RPI::Material::ChangeId m_previousChangeId = AZ::RPI::Material::DEFAULT_CHANGE_ID;
+
+        // Surface tags not used by default material
+        SurfaceData::SurfaceTag m_surfaceTag;
+        SurfaceData::SurfaceTag m_previousTag;
```

Comment:
I wonder if some of the changes are necessary in this class. I guess you probably made 2 classes earlier for default material and regular material (not sure how to call it).
It doesn't hurt to open inheritance. But the comments for the tags are not clear enough. Maybe pointing to TerrainSurfaceMaterialsListConfig for better understanding, or move the def to inside TerrainSurfaceMaterialsListConfig.

Reply:
Thanks for the callout - I forgot to revert some changes after moving back to using a single class for this, I'll remove the virtual destructor and add "final" back.

Inheritance doesn't really work well for this problem because it creates other issues. It would make logical sense to have a base class that didn't include the surface tags, then a subclass with the surface tags, but then you can't control the edit context layout - it will always show the parent class's edit context first then the child class. This would reverse the order of the material and surface tag in the UI. There's also some concerns over backwards compatibility of separating the fields in the serialize context across two classes.